src/utils/timing.test.ts

```diff
@@ -1,55 +1,9 @@
 import * as assert from "assert";
 import "mocha";
 import * as sinon from "sinon";
-import { IntervalPoller, pauseWithJitter } from "./timing";
+import { IntervalPoller, pauseWithJitter, raceWithTimeout } from "./timing";
 
-/** mocha tests over function pauseWithJitter */
-describe("pauseWithJitter", () => {
-  it("should pause", async () => {
-    const clock = sinon.useFakeTimers();
-
-    try {
-      const minMs = 20;
-      const maxMs = 50;
-
-      // add callback to track when pauseWithJitter resolves
-      let resolved = false;
-      const promise = pauseWithJitter(minMs, maxMs).then(() => {
-        resolved = true;
-      });
-
-      // shouldn't resolve yet
-      assert.strictEqual(resolved, false, "resolved immediately");
-      // fast forward just before minMs
-      const firstJump = minMs - 1;
-      await clock.tickAsync(firstJump);
-      assert.strictEqual(resolved, false, "resolved too early");
-
-      // fast forward past the maxMs time; should now be resolved
-      await clock.tickAsync(maxMs - firstJump + 1);
-      await promise;
-      assert.strictEqual(resolved, true, "didn't resolve before maxMs time");
-    } finally {
-      clock.restore();
-    }
-  });
-
-  it("should throw on invalid inputs", () => {
-    assert.rejects(async () => {
-      await pauseWithJitter(-1, 100);
-    });
-    assert.rejects(async () => {
-      await pauseWithJitter(100, -1);
-    });
-    assert.rejects(async () => {
-      await pauseWithJitter(100, 99);
-    });
-  });
-});
-
-/** mocha tests over class IntervalPoller */
-
-describe("IntervalPoller", () => {
+describe("utils/timing", () => {
```

Comment:
New toplevel describe block for this suite, which then makes the diff look strange.

New sub-block for tests needing sinon.SinonFakeTimers, holding the preexisting tests for `pauseWithJitter` and new tests for `raceWithTimeout`.